### ISSUE TITLE
Don't warn on empty reductions

### ISSUE THREAD

**Issue by max-sixty** [pydata/xarray#3811]
Numpy warns when computing over an all-NaN slice. We handle that case reasonably and so should handle and discard the warning.

#### MCVE Code Sample

```python
In [1]: import xarray as xr                                                                                                                                         

In [2]: import numpy as np                                                                                                                                          

In [3]: da = xr.DataArray(np.asarray([np.nan]*3))                                                                                                                   

In [4]: da                                                                                                                                                          
Out[4]: 
<xarray.DataArray (dim_0: 3)>
array([nan, nan, nan])
Dimensions without coordinates: dim_0

In [6]: da.mean()                                                                                                                                                   
[...]/python3.6/site-packages/xarray/core/nanops.py:142: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis=axis, dtype=dtype)
Out[6]: 
<xarray.DataArray ()>
array(nan)

```

#### Expected Output

No warning

#### Problem Description

Somewhat discussed in https://github.com/pydata/xarray/issues/1164, and https://github.com/pydata/xarray/issues/1652, but starting a separate issue as it's more important than just noise in the test suite, and not covered by the existing work on comparisons & arithmetic

#### Output of ``xr.show_versions()``
<details>

</details>

**Comment by max-sixty** [comment]
As discussed in the team meeting, there's consensus these should be silenced!

**Comment by keewis** [comment]
should have been closed by #4369
